File: app/src/main/java/com/designpatterns/demo/behavioral/chainofresponsibility/java/Postman.java
package com.designpatterns.demo.behavioral.chainofresponsibility.java;

/**
 * 创建抽象处理者类 快递员抽象类
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/6/5 上午11:53
 * @project_name DesignPatternsDemo
 */
public abstract class Postman {

    public Postman nextPostman;

    public abstract void handleCourier(String adress);

}

File: app/src/main/java/com/designpatterns/demo/behavioral/chainofresponsibility/kotlin/Postman.kt
package com.designpatterns.demo.behavioral.chainofresponsibility.kotlin

/**
 *
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/6/5 下午1:52
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
abstract class Postman{

    var nextPostman:Postman?=null

    abstract fun handleCourier(adress:String)


}